michael/express/lab03/routes/items.js

```diff
@@ -0,0 +1,50 @@
+const { Router } = require("express");
+const Item = require("../models/Item");
+const jwtMiddleware = require("../helpers/jwt.middleware");
+
+const itemsRouter = Router();
+
+itemsRouter.get("/", [jwtMiddleware], async (req, res) => {
+	try {
+		const items = await Item.find().populate("list");
+		res.send(items);
+	} catch (err) {
+		console.log(err);
+	}
+});
+
+itemsRouter.post("/", [jwtMiddleware], async (req, res) => {
+	try {
+		const item = await Item(req.body);
+		await item.save();
+		res.send(item);
+	} catch (err) {
+		console.log(err);
+	}
+});
+
+itemsRouter.patch("/:id", [jwtMiddleware], async (req, res) => {
+	try {
+		const item = await Item.findByIdAndUpdate({ _id: req.params.id });
+		await item.set(req.body);
+		await item.save();
+		res.send(item);
+	} catch (err) {
+		console.log(err);
+	}
+});
+
+itemsRouter.delete("/:id", [jwtMiddleware], async (req, res) => {
```

Comment:
You are making sure the current user is logged in, which is great, but we also need to check to see if the logged in user is the same user who created the item, otherwise anyone who is logged in can delete anybody else's items.